commons/plasma_sample_data_basic_query.R

```diff
@@ -0,0 +1,198 @@
+#'
+#' @return A date object representing the approximate event date.
+#'
+#' @examples
+#' calc_approx_date("2000-01-01", 20.5)
+#' calc_approx_date(as.Date("2000-01-01"), 20.5)
+#'
+#' @importFrom lubridate ymd years days
+#' @export
+calc_approx_date <- function(date, event_age) {
+  if (is.character(date)) {
+    date <- ymd(date)
+  }
+  int_p <- floor(event_age)
+  fract_p <- event_age - int_p
+  approx_days <- round(fract_p * 365.25)
+  approx_event_date <- date + years(int_p) + days(approx_days)
+  return(approx_event_date)
+}
+
+# Gather some sample annotations
+
+sql <- paste0(
+  "SELECT * ",
+  "FROM finngen-production-library.",
+  "sandbox_tools_r12.",
+  "minimum_extended_r12_v1 ",
+  "AS A ",
+  "LEFT JOIN (",
+  "SELECT IID, batch ",
+  "FROM finngen-production-library.",
+  "sandbox_tools_r12.",
+  "covariates_r12_v1)",
+  "AS B ",
+  "ON A.FINNGENID = B.IID"
+)
+anno <- as.data.table(bq_table_download(bq_project_query(projectid, sql)))
+anno[, IN_DF12_ANALYSIS := ! is.na(batch)]
+
+# Read in plasma sample availability data
+
+F10_1 <- fread("/finngen/library-red/task_force_data/plasma_availability/F10/data/F10_1.txt")
+F10_2 <- fread("/finngen/library-red/task_force_data/plasma_availability/F10/data/F10_2.txt")
+F06 <- fread("/finngen/library-red/task_force_data/plasma_availability/F06_F07_v2/data/F06_v2.txt")
+F07 <- fread("/finngen/library-red/task_force_data/plasma_availability/F06_F07_v2/data/F07_v2.txt")
+BS <- fread("/finngen/library-red/task_force_data/plasma_availability/blood_service_plasma_sample_availability/data/blood_service_plasma_sample_availability.txt")
+
+setnames(BS, "BIOBANK", "BIOBANK_PLASMA")
+
+RR <- rbind(F06, F07, F10_1, F10_2, BS, fill = T)
+RR[, COHORT_FINNGENID := NULL] # COHORT_FINNGENID not set for all data so remove and add later from annotation
+RR <- RR[!is.na(FINNGENID)]
+RR <- unique(RR)
+RR[, FINNGENID_N_PLASMA_SAMPLES := .N, by = .(FINNGENID)]
+RR[, SERIAL_SAMPLES := FINNGENID_N_PLASMA_SAMPLES > 1]
+RR[, HOURS_FROM_COLLECTION_TO_FREEZING := round(time_length(difftime(APPROX_TIMESTAMP_FREEZING, APPROX_TIMESTAMP_COLLECTION), "hours"), 2)]
+RR[, HOURS_FROM_COLLECTION_TO_FREEZING_BIN := cut(HOURS_FROM_COLLECTION_TO_FREEZING, breaks = c(0, 4, 6, 8, Inf), labels = c("<4h", "<6h", "<8h", ">8h"))]
+RR <- merge(RR, anno, by = "FINNGENID", all = T)
+RR[is.na(AVAILABILITY), ":="(AVAILABILITY = "NO", FINNGENID_N_PLASMA_SAMPLES = 0)]
+RR[, BIOBANK := gsub("^THL BIOBANK.*", "THL BIOBANK", COHORT)]
+RR[, BIOBANK := gsub("^ARCTIC BIOBANK.*", "ARCTIC BIOBANK", BIOBANK)]
+
+# Get phenotype data
+
+endpoints <- strsplit(c(
+"K11_KELAIBD
+C3_CANCER_WIDE
+G6_MS
+G6_AD_WIDE
+G6_PARKINSON
+G6_ALS
+K11_FIBROCHIRLIV
+H7_AMD
+ILD_ENDPOINTS
+I9_HEARTFAIL
+M13_ANKYLOSPON
+M13_SLE
+M13_RHEUMA
+M13_PSORIARTH
+M13_SACROILIITIS
+M13_SYSTSLCE
+N14_RENFAIL
+J10_ASTHMA
+J10_COPD
+N14_ACUTERENFAIL
+N14_CHRONKIDNEYDIS
+DM_RETINOPATHY
+L12_ATOPIC"), split = "\n")[[1]]
+
```

Comment:
no need to do for this example necessarily but could add these general things as functions so the main logic is easy to follow and not get tangled on  how exactly some things are fetched and renamed.......   e.g. get_endpoint_events( eplist). Model for how we could start adding common functionalities into separate modules easily. Could actually have broader discussion again about what pheno team has developed already.